src/main/java/core/basesyntax/PurchaseOperation.java

```diff
@@ -0,0 +1,10 @@
+package core.basesyntax;
+
+import java.util.Map;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction, Map<String, Integer> storage) {
+        storage.put(transaction.getFruit(), storage.getOrDefault(transaction.getFruit(), 0) - transaction.getQuantity());
+    }
```

Comment:
Handling Purchase operation requires checking the result balance before saving it in the Storage to ensure it is positive. If the balance after the purchase would be negative, you should throw a RuntimeException to prevent invalid transactions.